File: src/main/java/macro303/keschet/pieces/Emperor.java
package macro303.keschet.pieces;

import macro303.keschet.Colour;
import macro303.keschet.Direction;
import org.jetbrains.annotations.NotNull;

import static macro303.keschet.Direction.*;

/**
 * Created by Macro303 on 2018-02-08.
 */
public class Emperor extends Piece {
	@NotNull
//	private static final String symbol = "\u265A";
	private static final String symbol = "E";
	private static final int maxDistance = 4;
	@NotNull
	private static final Direction[] validDirections = new Direction[]{NORTH, NORTH_EAST, EAST, SOUTH_EAST, SOUTH, SOUTH_WEST, WEST, NORTH_WEST};

	public Emperor(@NotNull Colour teamColour) {
		super(teamColour, maxDistance, symbol, validDirections);
	}

	public Emperor() {
		super(maxDistance, symbol, validDirections);
	}

	@Override
	public String toString() {
		return "Emperor{}";
	}
}

File: src/main/kotlin/macro303/keschet/pieces/Emperor.kt
package macro303.keschet.pieces

import macro303.keschet.Colour
import macro303.keschet.Direction
import java.util.*

internal data class Emperor(override var teamColour: Colour) : IPiece {
	override val maxDistance = 4
	override val symbol = "E"
	override val validDirections = arrayOf(Direction.NORTH, Direction.NORTH_EAST, Direction.EAST, Direction.SOUTH_EAST, Direction.SOUTH, Direction.SOUTH_WEST, Direction.WEST, Direction.NORTH_WEST)

	override fun toString(): String {
		return "Emperor(teamColour=$teamColour, maxDistance=$maxDistance, symbol='$symbol', validDirections=${Arrays.toString(validDirections)})"
	}
}